### PR TITLE
[#IOPID-256] upgrade xmldom

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* better check for multiple saml Responses
* migrated `xmldom` dependency to `@xmldom/xmldom`
* removed custom type definitions for `xmldom`

#### Motivation and Context

removing custom xmldom types definitions in this library and align the dependency to its latest release

relevant discussion: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/65454

**NOTE from the library author**:
> a lot of time has already been invested (and has to be invested, no ETA) into version 0.9.0, which is currently available as a beta version with the next tag, for which I'm planning to throw an error when no Document is returned. (This is not spec compliant, but I think it is still better than returning undefined.) Currently that next version provides a type definition that indicate that it might return undefined, which is going to change before 0.9.0 is released as latest.

in the current implementation of the library we have a tryCatch on the `parseFromString` method. The future upgrade to `0.9.x` shouldn't require a major refactor(maybe only unit tests).

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/utils/saml.ts [hunk 1]
```diff
@@ -156,17 +156,26 @@ export const getPreValidateResponse =
...
-        E.chain(
-          E.fromNullable(
-            new Error("Missing Response element inside SAML Response")
-          )
+        E.chainFirst(
+          // updated versions of xmldom will convert any additional root node in a text node(https://github.com/advisories/GHSA-crh6-fp67-6883)
+          // to ensure if more than one samlp:Response node was provided, we must check if the sibling node is present, returning an error afterwards
+          (item) =>
+            pipe(
+              item.nextSibling,
```
Comment (gunzip) [inline src/utils/saml.ts:169]: maybe using `fromPredicate` we can avoid to swap and make this more readable?

Comment (arcogabbo) [reply src/utils/saml.ts]: thank you for this suggestion, I applied it in https://github.com/pagopa/io-spid-commons/pull/147/commits/960fbbf1e0e15cd3c3f440aa648ff3227edd2f4e

File: src/utils/saml.ts [hunk 2]
```diff
@@ -156,16 +156,19 @@ export const getPreValidateResponse =
...
-        // this check is bound to the previous one, because we can receive no Response based on the official validator guidelines
-        E.chain(
-          E.fromNullable(
-            new Error("Missing Response element inside SAML Response")
+        E.chainFirst(
+          E.fromPredicate(
+            // updated versions of xmldom will convert any additional root node in a text node(https://github.com/advisories/GHSA-crh6-fp67-6883)
+            // to ensure if more than one samlp:Response node was provided, we must check if the sibling node is present, returning an error afterwards
+            (item) =>
+              item.nextSibling === null || item.nextSibling === undefined,
```
Comment (BurnedMarshal) [inline src/utils/saml.ts:169]:
```suggestion
            ({ nextSibling }) =>
              nextSibling === null || nextSibling === undefined,
```

**Review by BurnedMarshal — APPROVED** [review]
LGTM, only an optional stylish suggestion.

**Pull request merged**